gatsby-node.js

```diff
@@ -2,75 +2,82 @@ const path = require("path")
 const _ = require("lodash")
 const { createFilePath } = require("gatsby-source-filesystem")
 
-exports.createPages = ({ graphql, actions }) => {
+exports.createPages = async ({ graphql, actions, reporter }) => {
   const { createPage } = actions
-
-  return new Promise((resolve, reject) => {
-    const blogPost = path.resolve("./src/templates/blog-post.tsx")
-    const catePage = path.resolve("./src/templates/category-template.tsx")
-
-    resolve(
-      graphql(
-        `
-          {
-            allMarkdownRemark(
-              sort: { fields: [frontmatter___date], order: DESC }
-              limit: 1000
-            ) {
-              group(field: frontmatter___categories) {
-                fieldValue
+  const result = await graphql(
+    `
+      {
+        allMarkdownRemark(
+          sort: { fields: [frontmatter___date], order: DESC }
+          limit: 1000
+        ) {
+          categories: group(field: frontmatter___categories) {
+            fieldValue
+          }
+          authors: group(field: frontmatter___author) {
```

Comment:
I'm curious how `group()` queries actually work. What does this do? I realize this sounds like a silly question after all this but I hadn't thought about it until now. Does it come from gatsby?

Reply:
It performs like `GROUP BY` in SQL. Declare the filed, then query the fieldValue. The most important thing is the result is **distinct**. When we want to create the category page, it is super useful. In this page, they introduce the group, https://www.gatsbyjs.org/docs/graphql-reference/#group .